### DISCUSSION TITLE
Set iTerm2 tab & window title w/o flickering

### DISCUSSION THREAD

**Discussion by espresso1736** [xonsh/xonsh#5695] [Q&A]
I can set the tab and window title the way I want it via a custom field in $PROMPT, however before it shows my custom title, it briefly flashes the "standard" title.  I assume this comes from some part of xonsh, because this doesn't happen under bash.  I am using the term_integration xontrib, in case that matters.  Any pointers as to how to fix this will be appreciated.  Thanks.

**Comment by anki-code** [comment]
What if you set the $PROMPT just in `~/.xonshrc`? 
If it works better please create an issue in the [xontrib-term-integrations](https://github.com/jnoortheen/xontrib-term-integrations) repostiory.

**Reply by espresso1736** [reply]
Thanks for the suggestion.  The $PROMPT is already only being set in ~/.xonshrc.  I had it before the load of term_integration, but have tried it after the load, and also commenting out the load - neither made any difference that I could see.  Must not be related to term_integration.
